presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1171 @@
+    {
+        // Check all the column types, and throw an exception if the types of a map are complex
+        // While it is a rare case, this is not supported by the Accumulo connector
+        ImmutableSet.Builder<String> columnNameBuilder = ImmutableSet.builder();
+        for (ColumnMetadata column : meta.getColumns()) {
+            if (Types.isMapType(column.getType())) {
+                if (Types.isMapType(Types.getKeyType(column.getType()))
+                        || Types.isMapType(Types.getValueType(column.getType()))
+                        || Types.isArrayType(Types.getKeyType(column.getType()))
+                        || Types.isArrayType(Types.getValueType(column.getType()))) {
+                    throw new PrestoException(VALIDATION,
+                            "Key/value types of a MAP column must be plain types");
+                }
+            }
+
+            columnNameBuilder.add(column.getName().toLowerCase(Locale.ENGLISH));
+        }
+
+        // Validate the columns are distinct
+        if (columnNameBuilder.build().size() != meta.getColumns().size()) {
+            throw new PrestoException(VALIDATION, "Duplicate column names are not supported");
+        }
+
+        Optional<Map<String, Pair<String, String>>> columnMapping = AccumuloTableProperties.getColumnMapping(meta.getProperties());
+        if (columnMapping.isPresent()) {
+            // Validate there are no duplicates in the column mapping
+            long distinctMappings = columnMapping.get().values().stream().distinct().count();
+            if (distinctMappings != columnMapping.get().size()) {
+                throw new PrestoException(VALIDATION,
+                        "Duplicate column family/qualifier pair detected in column mapping, check the value of " +
+                                AccumuloTableProperties.COLUMN_MAPPING);
+            }
+
+            // Validate no column is mapped to the reserved entry
+            String reservedRowIdColumn = AccumuloPageSink.ROW_ID_COLUMN.toString();
+            if (columnMapping.get().values().stream()
+                    .filter(pair -> pair.getKey().equals(reservedRowIdColumn)
+                            && pair.getValue().equals(reservedRowIdColumn)).count() > 0) {
+                throw new PrestoException(VALIDATION,
+                        format("Column familiy/qualifier mapping of %s:%s is reserved",
+                                reservedRowIdColumn, reservedRowIdColumn));
+            }
+        }
+        else if (AccumuloTableProperties.isExternal(meta.getProperties())) {
+            // Column mapping is not defined (i.e. use column generation) and table is external
+            // But column generation is for internal tables only
+            throw new PrestoException(VALIDATION,
+                    "Column generation for external tables is not supported, must specify " +
+                            AccumuloTableProperties.COLUMN_MAPPING);
+        }
```

Comment:
Is it possible to not have column mapping in table properties when it's internal table?
E.g. we don't fall into first `if (columnMapping.isPresent())`, nor we have external table?